src/components/markdown-renderer/replace-components/csv/csv-table.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+
+export interface CsvTableProps {
+  index: number
+  code: string
+  delimiter: string
+  showHeader: boolean
+  tableRowClassName?: string
+  tableColumnClassName?: string
+}
+
+const parseCsvToRowsAndColumn = (csvText: string, csvColumnDelimiter: string): string[][] => {
+  const rows = csvText.split('\n');
+  if (!rows || rows.length === 0) {
+    return [];
+  }
+
+  return rows.map(row => row.split(csvColumnDelimiter));
+}
+
+export const CsvTable: React.FC<CsvTableProps> = ({ index, code, delimiter, showHeader, tableRowClassName, tableColumnClassName}) => {
```

Comment:
You're calculating the CSV rows and columns on every render, even if the props didn't change. Please add some useMemo-Calls to cache results.

Reply:
done